SOD.Common/Helpers/PluginDetection.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using System.Linq;
+using BepInEx;
+using BepInEx.Configuration;
+using BepInEx.Unity.IL2CPP;
+using UniverseLib;
+
+namespace SOD.Common.Helpers
+{
+    public sealed class PluginDetection
+    {
+        /// <summary>
+        /// Raised right after the BepInEx chainloader finishes loading the
+        /// last plugin.
+        /// </summary>
+        public event EventHandler OnAllPluginsFinishedLoading;
+
+        /// <summary>
+        /// True if the BepInEx chainloader has finished loading all plugins,
+        /// false otherwise (is set to true after OnAllPluginsFinishedLoading).
+        /// </summary>
+        public bool AllPluginsFinishedLoading { get; private set; } = false;
+
+        private bool _initialized = false;
+
+        internal PluginDetection()
+        {
+            Initialize();
+        }
+
+        internal void Initialize()
+        {
+            if (_initialized)
+            {
+                return;
+            }
+            _initialized = true;
+
+            // Hook up the OnAllPluginsFinishedLoading event and related
+            var invokeEventAction = () => OnAllPluginsFinishedLoading?.Invoke(this, EventArgs.Empty);
```

Comment:
Put it into a private method and just assign the method as such:

` IL2CPPChainloader.Instance.Finished += OnPluginLoadFinished;`

```
private void OnPluginLoadFinished()
{
    AllPluginsFinishedLoading = true;
    OnAllPluginsFinishedLoading?.Invoke(this, EventArgs.Empty);
    IL2CPPChainloader.Instance.Finished -= OnPluginLoadFinished;
}
```

Reply:
Agreed, will change